src/cache.ts

```diff
@@ -0,0 +1,82 @@
+import crypto from "node:crypto";
+import fs from "node:fs/promises";
+import path from "node:path";
+import { mkdirSync } from "node:fs";
+
+export class Cache {
+  private dir: string;
+  private loading: Record<string, Promise<unknown>>;
+  private isDisabled: boolean;
+
+  constructor(dir: string, isDisabled = false) {
+    this.dir = dir;
+    this.loading = {};
+    this.isDisabled = isDisabled;
+
+    if (!this.isDisabled) {
+      mkdirSync(this.dir, { recursive: true });
```

Comment:
This will block the entire process. Since it's not possible to make a constructor async, an alternative would be to call the async version lazily at the point where the directory is needed.